snakemake/common/prefix_lookup.py

```
@@ -0,0 +1,53 @@
+                # Exit loop, if iteration has reached "" query or key
+                break
+
+            query = self._common_prefix(k, query)
+
+    def _common_prefix(self, s1: str, s2: str) -> str:
+        """Utility function that gives the common prefix of two strings.
+
+        Except, if the strings are identical, returns the string minus one letter.
+        Behaviour is undefined if either string is empty.
+        """
+        for i, (l1, l2) in enumerate(zip(s1, s2)):
```

Ruff (0.8.2)

Loop control variable `i` not used within loop body

(B007)